### PR TITLE
Release/5.0.0

### PR DESCRIPTION
**Pull request by skhozin**
Updated README with the latest SDK version and TOC
Uploaded JAVA8-MIGRATION-GUIDE.md file

### CODE REVIEW

File: JAVA8-MIGRATION-GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Java 8 Migration
+
+The Thunderhead SDK 5.0.0+ targets the Java8 Language Specification.  This requires that consuming applications also target Java 8.
+
+#### Affects
+
+All applications previously integrated with the Thunderhead SDK <= 5.0.0 that were not already targeting Java 8 or
```
Comment (tothstephen) [inline JAVA8-MIGRATION-GUIDE.md:7]: `previously` isn't needed here as new implementations are also affected by the change.
`All applications integrating the Thunderhead SDK 5.0.0 that are not already targeting Java 8 or`

Comment (skhozin) [reply JAVA8-MIGRATION-GUIDE.md]: Updated.

Comment (tothstephen) [reply JAVA8-MIGRATION-GUIDE.md]: No quite , the second highlighted text in previous comment more accurately describes the affected audience.  No need for `<=` at all and the tense changes when not referencing older SDK versions.

Comment (skhozin) [reply JAVA8-MIGRATION-GUIDE.md]: Updated.

File: JAVA8-MIGRATION-GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Java 8 Migration
+
+The Thunderhead SDK 5.0.0+ targets the Java8 Language Specification.  This requires that consuming applications also target Java 8.
```
Comment (andreipop) [inline JAVA8-MIGRATION-GUIDE.md:3]: please change `Java8 Language Specification` to `Java 8 Language Specification`

File: JAVA8-MIGRATION-GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+       compileOptions {
+           sourceCompatibility 1.8
+           targetCompatibility 1.8
+       }
+   }
+```
+
+#### Steps to fix
+
+Please place the recommended code into your _app_ level `build.gradle` file in the `android` configuration closure as shown above or in the error message.
```
Comment (andreipop) [inline JAVA8-MIGRATION-GUIDE.md:22]: have we actually tried this and this is all that's needed? https://developer.android.com/studio/write/java8-support suggests that there may be a few more steps.

File: README.md [hunk 4]
```diff
@@ -298,7 +370,9 @@ By disabling automatic Interaction detection, the SDK won’t automatically send
 
 You can set this back to false at any point to restart automatic Interaction detection.
 
-### Send an Interaction request programmatically 
+### Send Interactions to Thunderhead ONE or Salesforce Interaction Studio
```
Comment (andreipop) [inline README.md:373]: please drop `to Thunderhead ONE or Salesforce Interaction Studio`

File: README.md [hunk 5]
```diff
@@ -399,25 +473,25 @@ private void showVariants() {
 }
 ```
 ### Send Properties to Thunderhead ONE or Salesforce Interaction Studio
```
Comment (andreipop) [inline README.md:475]: please drop `to Thunderhead ONE or Salesforce Interaction Studio`

File: TROUBLESHOOTING-GUIDE.md [hunk 6]
```diff
@@ -44,6 +44,7 @@ For more information backing up user data see [Back up user data](https://develo
 in the Android Developer's guide
 
 ### Resolving > The library com.google.firebase:firebase-iid is being requested by various other libraries
```
Comment (andreipop) [inline TROUBLESHOOTING-GUIDE.md:46]: could we add a TOC to this doc as well please

File: JAVA8-MIGRATION-GUIDE.md (hunk 3 above)
Comment (skhozin) [reply JAVA8-MIGRATION-GUIDE.md:22]: We can add something like: `For more information visit [Java 8 support]( https://developer.android.com/studio/write/java8-support) page.`?

File: TROUBLESHOOTING-GUIDE.md (hunk 6 above)
Comment (skhozin) [reply TROUBLESHOOTING-GUIDE.md:46]: @alexandern17 Could you please take care of it?

File: JAVA8-MIGRATION-GUIDE.md (hunk 2 above)
Comment (skhozin) [reply JAVA8-MIGRATION-GUIDE.md]: Updated.

File: JAVA8-MIGRATION-GUIDE.md (hunk 3 above)
Comment (skhozin) [reply JAVA8-MIGRATION-GUIDE.md:21]: Added a link to Java 8 support page. Please let me know if this is not what we wanted.

File: README.md (hunk 4 above)
Comment (skhozin) [reply README.md]: Dropped.

File: README.md (hunk 5 above)
Comment (skhozin) [reply README.md]: Dropped.

**Pull request merged**